server/src/data_sources/memory_provider.rs

```diff
@@ -51,11 +51,22 @@ impl MemoryProvider {
             self.token_store.insert(token.token.clone(), token.clone());
             if token.token_type == Some(crate::types::TokenType::Client) {
                 self.tokens_to_refresh
-                    .insert(token.token.clone(), FeatureRefresh::new(token));
+                    .entry(token.clone().token)
+                    .or_insert(FeatureRefresh::new(token.clone()));
             }
         }
+        self.reduce_tokens_to_refresh();
+    }
+    fn reduce_tokens_to_refresh(&mut self) {
+        let tokens: Vec<EdgeToken> = self
+            .tokens_to_refresh
+            .values()
+            .map(|r| r.token.clone())
+            .collect();
+        let minimized = crate::tokens::simplify(&tokens);
```

Comment:
I think it's time to remove:
https://github.com/Unleash/unleash-edge/blob/ca0a50d711f8c504f2ad9671929abc663639264b/server/src/tokens.rs#L3